pkg/liqonet/ipam/ipam_suite_test.go

```diff
@@ -1,13 +1,203 @@
+			ExternalCIDR: "10.0.1.0/24",
+		},
+		Status: netv1alpha1.TunnelEndpointStatus{
+			LocalPodCIDR:          "192.168.0.0/24",
+			LocalNATPodCIDR:       "192.168.1.0/24",
+			RemoteNATPodCIDR:      "10.0.70.0/24",
+			LocalExternalCIDR:     "192.168.3.0/24",
+			LocalNATExternalCIDR:  "192.168.4.0/24",
+			RemoteNATExternalCIDR: "192.168.5.0/24",
+		},
+	}
 )
 
 func TestIpam(t *testing.T) {
 	RegisterFailHandler(Fail)
 	RunSpecs(t, "Ipam Suite")
 }
+
+var _ = BeforeSuite(func() {
+	// Create custom network namespace for natmapping-operator.
+	iptNetns, err = netns.CreateNetns(iptNetnsName)
+	Expect(err).To(BeNil())
+
+	// Config IPTables for remote clusters
+	err = initIPTables()
+	Expect(err).To(BeNil())
+
+	// Launch natmapping operator
+	err = initNatMappingController()
+	Expect(err).To(BeNil())
+
+	// Init Ipam
+	err = initIpam()
+	Expect(err).To(BeNil())
+})
+
+var _ = AfterSuite(func() {
+	err := envTest.Stop()
+	Expect(err).To(BeNil())
+	err = terminateIPTables()
+	Expect(err).To(BeNil())
+	err = iptNetns.Close()
+	Expect(err).To(BeNil())
+})
+
+func initNatMappingController() error {
+	err = netv1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		return err
+	}
+	envTest = &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "deployments", "liqo", "crds")},
+	}
+	config, err := envTest.Start()
+	if err != nil {
+		return err
+	}
+	mgr, err := controllerruntime.NewManager(config, controllerruntime.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0",
+	})
+
+	controller, err = tunneloperator.NewNatMappingController(mgr.GetClient(), &readyClustersMutex, readyClusters, iptNetns)
+	if err != nil {
+		return err
+	}
+	go func() {
+		if err = mgr.Start(context.Background()); err != nil {
+			panic(err)
+		}
+	}()
+	dynClient = dynamic.NewForConfigOrDie(mgr.GetConfig())
+	k8sClient = mgr.GetClient()
+	return controller.SetupWithManager(mgr)
+}
+
+func initIpam() error {
+	ipamInt = liqonetIpam.NewIPAM()
+	n, err := rand.Int(rand.Reader, big.NewInt(2000))
+	if err != nil {
+		return err
+	}
+	err = ipamInt.Init(liqonetIpam.Pools, dynClient, 2000+int(n.Int64()))
+	if err != nil {
+		return err
+	}
+	// Set home cluster PodCIDR and ExternalCIDR
+	err = ipamInt.SetPodCIDR(homePodCIDR)
+	if err != nil {
+		return err
+	}
+	homeExternalCIDR, err = ipamInt.GetExternalCIDR(uint8(24))
+	if err != nil {
+		return err
+	}
+	// Assign networks to clusterID1
+	_, _, err = ipamInt.GetSubnetsPerCluster(remotePodCIDR, remoteExternalCIDR, clusterID4)
+	if err != nil {
+		return err
+	}
+	return ipamInt.AddLocalSubnetsPerCluster(localPodCIDR, localExternalCIDR, clusterID4)
+}
+
+func terminateIPTables() error {
+	err := iptNetns.Do(func(nn ns.NetNS) error {
+		var err error
+		err = ipt.RemoveIPTablesConfigurationPerCluster(tep)
+		if err != nil {
+			return err
+		}
+		err = ipt.Terminate()
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	return err
+}
+
+func initIPTables() error {
```

Comment:
I would refer to the semantic component in Liqo instead of the technology used under the hoods.